src/modeler/AutoRest.Swagger/Validation/ListByOperationsValidation.cs

```diff
@@ -0,0 +1,163 @@
+                    if (!(ListByRegex.IsMatch(pair.Value.OperationId) || pair.Value.OperationId.ToLower().EndsWith("_list")))
+                    {
+                        return null;
+                    }
+
+                    // if the operation does not return an array type or is not an xms pageable type, skip
+                    if (!(ValidationUtilities.IsXmsPageableOrArrayResponseOperation(pair.Value, serviceDefinition)))
+                    {
+                        return null;
+                    }
+
+                    // if all conditions pass, return the operation id
+                    return pair.Value.OperationId;
+                }).Where(opId=> opId!=null);
+
+                // if there are no operations matching our conditions, skip
+                if (IsNullOrEmpty(listOpIds))
+                {
+                    continue;
+                }
+
+                // populate valid list operation names for given path
+                List<string> opNames = GetValidListOpNames(pathObj.Key);
+                
+                // if url does not match any of the predefined regexes, skip
+                if (opNames == null || opNames.Count==0)
+                {
+                    continue;
+                }
+
+                // find if there are any operations that violate the rule
+                var errOpIds = listOpIds.Where(opId => !opNames.Contains(opId.ToLower()));
+                
+                // no violations found, skip
+                if (IsNullOrEmpty(errOpIds))
+                {
+                    continue;
+                }
+                // aggregate suggested op names in a single readable string for the formatter
+                var suggestedNames = opNames.Aggregate((curr, next)=> curr+", "+ next);
+                foreach (var errOpId in errOpIds)
+                {
+                    var stringParams = new List<string>() { errOpId };
+                    stringParams.AddRange(opNames);
+                    yield return new ValidationMessage(new FileObjectPath(context.File, context.Path), this, errOpId, suggestedNames);
+                }
+            }
+        }
+
+        private List<string> GetValidListOpNames(string path)
+        {
+            var opList = new List<string>();
+            var methodGrp = GetMethodGroup(path);
+            if (methodGrp == string.Empty || !SubscriptionRegex.IsMatch(path))
+            {
+                return opList;
+            }
+
+            // construct _listbysubscriptionid
+            opList.Add(string.Format(ListByTemplate, methodGrp, SubscriptionId).ToLower());
+            opList.Add(string.Format(ListTemplate, methodGrp).ToLower());
+
+            // construct _listbyresourcegroup
+            if ((RgRegex.IsMatch(path)))
+            {
+                opList.Add(string.Format(ListByTemplate, methodGrp, ResourceGroup).ToLower());
+            }
+            else
+            {
+                return opList;
```

Comment:
Yes, it is a cascading pattern, if the url path does not follow the ```/resourcegroup/{resourcegroupname}``` pattern then it does not follow the pattern ```/resourcegroup/{resourcegroupname}/providers/{namespace}/{type}/{type_name}```